tests/test_cli.py

```diff
@@ -0,0 +1,145 @@
+            '-c', "/config_folder/config_file.yml",
+            '-q',
+            '{"dataseturi__uri__contains": "osisaf", "source__instrument__short_name__icontains": "AMSR2"}',
+            ]
+
+    def test_extract_arg(self):
+        """shall return the correct argument values based on the 'sys.argv' """
+        arg = cli_download.parse_args()
+        self.assertEqual(arg.begin, '200')
+        self.assertEqual(arg.config_file, '/config_folder/config_file.yml')
+        self.assertEqual(arg.down_dir, '/test_folder/%Y_nh_polstere')
+        self.assertEqual(arg.end, '2020-08-22')
+        self.assertEqual(arg.geometry, 'POLYGON ((-22 84, -22 74, 32 74, 32 84, -22 84))')
+        self.assertEqual(arg.number_per_day, '100')
+        self.assertEqual(
+            arg.query,
+            '{"dataseturi__uri__contains": "osisaf", "source__instrument__short_name__icontains": "AMSR2"}')
+        # testing the flag performance
+        self.assertTrue(arg.rel_time_flag)
+        self.assertTrue(arg.use_filename_prefix)
+        sys.argv.remove('-r')
+        sys.argv.remove('-p')
+        arg = cli_download.parse_args()
+        self.assertFalse(arg.rel_time_flag)
+        self.assertFalse(arg.use_filename_prefix)
+
+    @mock.patch('geospaas_processing.downloaders.DownloadManager.__init__', return_value=None)
+    @mock.patch('geospaas_processing.downloaders.DownloadManager.download')
+    def test_correct_call_json_deserializer(self, mock_downloadmethod, mock_downloadmanagerinit):
+        """'json.loads' shall deserializer the whole string that comes after '-q' """
+        arg = cli_download.parse_args()
+        with mock.patch('json.loads') as mock_json:
+            cli_download.main(arg)
+        self.assertIn((
+            '{"dataseturi__uri__contains": "osisaf", "source__instrument__short_name__icontains": "AMSR2"}',),
+            mock_json.call_args)
+
+    @mock.patch('geospaas_processing.downloaders.DownloadManager.__init__', return_value=None)
+    @mock.patch('geospaas_processing.downloaders.DownloadManager.download')
+    def test_lack_of_calling_json_deserializer_when_no_query_appears(
+        self, mock_downloadmethod, mock_downloadmanagerinit):
+        """'json.loads' should not called when nothing comes after '-q' """
+        arg = cli_download.parse_args()
+        arg.query=""
+        with mock.patch('json.loads') as mock_json:
+            cli_download.main(arg)
+        self.assertIsNone(mock_json.call_args)
+
+    @mock.patch('geospaas_processing.downloaders.DownloadManager.__init__', return_value=None)
+    @mock.patch('geospaas_processing.downloaders.DownloadManager.download')
+    def test_correct_call_DownloadManager_without_file_prefix(
+            self, mock_downloadmethod, mock_downloadmanagerinit):
+        """shall return the proper call (with 'use_file_prefix': False) for the case of "
+        "lack of file prefix ('-p') in arguments"""
```

Comment:
the name of the method and the docstring don't seem to match.

Reply:
docstring is corrected. Done in the next commit.